pkg/cache/task_test.go

```diff
@@ -0,0 +1,301 @@
+
+	// new task
+	event0 := NewSimpleTaskEvent(task.applicationID, task.taskID, events.InitTask)
+	if err := task.handle(event0); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Pending)
+
+	// submit task to the scheduler-core
+	event1 := NewSubmitTaskEvent(app.applicationID, task.taskID)
+	if err := task.handle(event1); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Scheduling)
+
+	// allocated
+	event2 := NewAllocateTaskEvent(app.applicationID, task.taskID, string(pod.UID), "node-1")
+	if err := task.handle(event2); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Allocated)
+
+	// bound
+	event3 := NewBindTaskEvent(app.applicationID, task.taskID)
+	if err := task.handle(event3); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Bound)
+
+	// complete
+	event4 := NewSimpleTaskEvent(app.applicationID, task.taskID, events.CompleteTask)
+	if err := task.handle(event4); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Completed)
+}
+
+func TestTaskIllegalEventHandling(t *testing.T) {
+	mockedSchedulerApi := newMockSchedulerAPI()
+	mockedContext := initContextForTest()
+	resources := make(map[v1.ResourceName]resource.Quantity)
+	containers := make([]v1.Container, 0)
+	containers = append(containers, v1.Container{
+		Name: "container-01",
+		Resources: v1.ResourceRequirements{
+			Requests: resources,
+		},
+	})
+	pod := &v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name: "pod-resource-test-00001",
+			UID:  "UID-00001",
+		},
+		Spec: v1.PodSpec{
+			Containers: containers,
+		},
+	}
+
+	app := NewApplication("app01", "root.default",
+		"bob", map[string]string{}, mockedSchedulerApi)
+	task := NewTask("task01", app, mockedContext, pod)
+	assert.Equal(t, task.GetTaskState(), events.States().Task.New)
+
+	// new task
+	event0 := NewSimpleTaskEvent(task.applicationID, task.taskID, events.InitTask)
+	if err := task.handle(event0); err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Pending)
+
+	// verify illegal event handling logic
+	event2 := NewAllocateTaskEvent(app.applicationID, task.taskID, string(pod.UID), "node-1")
+	err := task.handle(event2)
+	if err == nil {
+		t.Fatal("expecting an error, event AllocateTask is illegal when task is Pending")
+	}
+	// task state should not have changed
+	assert.Equal(t, task.GetTaskState(), events.States().Task.Pending)
+}
+
+func TestReleaseTaskAllocation(t *testing.T) {
+	mockedSchedulerApi := newMockSchedulerAPI()
+	mockedContext := initContextForTest()
+	apiProvider := mockedContext.apiProvider
+	mockedApiProvider, ok := apiProvider.(*client.MockedAPIProvider)
+	if !ok {
+		t.Fatal("expecting MockedAPIProvider")
+	}
```

Comment:
we can use `assert.Assert(t, ok, "expecting MockedAPIProvider")`

Reply:
done